### PR TITLE
Protect: Refactor AdminSectionHero

### PR DESCRIPTION
**Pull request by nateweller**
## Proposed changes:

* Changes Protect's `AdminSectionHero` component to export sub-components, replacing the `main` and `secondary` props with `AdminSectionHero.Main` and `AdminSectionHero.Aside`.
```jsx
// old implementation
<AdminSectionHero
    main={ (
        <>
            <AdminSectionHero.Heading showIcon variant="warning">
                Hello, World!
            </AdminSectionHero.Heading>
            <AdminSectionHero.Subheading>
                Code is poetry.
            </AdminSectionHero.Subheading>
        </>
    ) }
    mainClassName="my-main-class-name"
    secondary={ <img src="example.jpg" /> }
    preserveSecondaryOnMobile={ true }
/>

// proposed implementation
<AdminSectionHero>
    <AdminSectionHero.Main className="my-main-class-name">
        <AdminSectionHero.Heading icon="warning">
            Hello, World!
        </AdminSectionHero.Heading>
        <Text>
            Code is poetry.
        </Text>
    </AdminSectionHero.Main>
    <AdminSectionHero.Aside>
        <img src="example.jpg" />
    </AdminSectionHero.Aside>
</AdminSectionHero>
```

* Adjust `AdminSectionHero` styling, use flexbox instead of grid.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack/pull/40317#issuecomment-2524048189

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Review the page headers in Jetpack Protect on various screen sizes and states.

## Screenshots

Screenshot 2024-12-08 at 7 46 30 PM

Screenshot 2024-12-08 at 7 46 41 PM

Screenshot 2024-12-08 at 7 47 25 PM

### CODE REVIEW

**Review by dkmyta — COMMENTED** [review]
Great improvements! Made note of one potential area for improvement and a minor nitpick.

In the `scanning` state at `1100px` the `inProgressAnimation` wraps to a second line. Previously the main section shrunk to fit to a point when the animation became hidden. If thats no longer an option maybe we can just hide it earlier?
Screen Shot on 2024-12-09 at 10-28-26 (1)

Nit - should the `FirewallPage` status icon/label be spaced out further? Perhaps a similar margin (`16px`) as the `lastCheckedLocalTimestamp` in the `ScanPage` too keep things uniform?
Screen Shot on 2024-12-09 at 10-24-24 (1)

File: projects/plugins/protect/src/js/routes/firewall/firewall-admin-section-hero.tsx [hunk 1]
```diff
@@ -84,16 +84,18 @@ const FirewallAdminSectionHero = () => {
...
-					<Status status={ 'on' === status ? 'active' : 'inactive' } label={ statusLabel } />
-					<AdminSectionHero.Heading>{ heading }</AdminSectionHero.Heading>
-					<AdminSectionHero.Subheading>{ subheading }</AdminSectionHero.Subheading>
-				</>
-			}
-			secondary={ wafSupported && <FirewallStatCards /> }
-		/>
+		<AdminSectionHero>
+			<AdminSectionHero.Main>
+				<Status status={ 'on' === status ? 'active' : 'inactive' } label={ statusLabel } mb={ 2 } />
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/firewall/firewall-admin-section-hero.tsx]: Looks like this was actually intended to have a 16px margin-bottom, but `mb` does not appear to be an assignable prop.

File: projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx [hunk 2]
```diff
@@ -84,94 +84,88 @@ const ScanAdminSectionHero: React.FC = () => {
...
+				<OnboardingPopover
+					id={ hasPlan ? 'paid-daily-and-manual-scans' : 'free-daily-scans' }
+					position={ isSm ? 'bottom' : 'middle right' }
+					anchor={ dailyScansPopoverAnchor }
+				/>
+				<AdminSectionHero.Heading icon={ numThreats > 0 ? 'error' : 'success' }>
+					{ numThreats > 0
+						? sprintf(
+								/* translators: %s: Total number of threats/vulnerabilities */
+								__( '%1$s active %2$s', 'jetpack-protect' ),
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx]: While we're at it here, might be a good opportunity to fix our threat(s)/vulnerability(ies) string concatenations.

File: projects/plugins/protect/src/js/routes/firewall/firewall-admin-section-hero.tsx (hunk 1 above)
Comment (nateweller) [reply projects/plugins/protect/src/js/routes/firewall/firewall-admin-section-hero.tsx]: Changed to using a custom style for the margin: https://github.com/Automattic/jetpack/pull/40516/commits/01586324b772b4be63a0b3e74f235074025f4c67 :+1:

File: projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx (hunk 2 above)
Comment (nateweller) [reply projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx]: Addressed via https://github.com/Automattic/jetpack/pull/40516/commits/a0d303d7d4bcf704ab7a73387047ba139bc40426

File: projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx [hunk 3]
```diff
@@ -59,6 +59,23 @@ const ScanAdminSectionHero: React.FC = () => {
 		lastCheckedLocalTimestamp = new Date( status.lastChecked + ' UTC' ).getTime();
 	}
 
+	let heading = __( "Don't worry about a thing", 'jetpack-protect' );
+	if ( numThreats > 0 ) {
+		if ( hasPlan ) {
+			heading = sprintf(
+				/* translators: %s: Total number of threats */
+				__( '%1$s active threats', 'jetpack-protect' ),
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx]: I think we'll want to use `_n` here for cases that this wont address (eg. `1 active threats`)? Similarly in the vulnerabilities translation.

Comment (nateweller) [reply projects/plugins/protect/src/js/routes/scan/scan-admin-section-hero.tsx]: Totally, fixed via https://github.com/Automattic/jetpack/pull/40516/commits/03dd6b3ee8339bf4bc218c2099bce3b7f386ae37 :+1:

**Review by dkmyta — APPROVED** [review]
LGTM

**Pull request merged**